### ISSUE TITLE
🐛 Extension flags lines with Biome lint errors when linting is disabled

### ISSUE THREAD

**Issue by lylejohnson-bw** [biomejs/biome-vscode#1031]
### VS Code version

1.116.0

### Extension version

3.6.1

### Biome version

2.4.12

### Operating system

- [ ] Windows
- [x] macOS
- [ ] Linux

### Description

I've disabled the linter in my root-level `biome.json` (my only `biome.json` file) but when I open a file that contains style issues the offending lines are flagged. I'd like to be able to disable linting by the extension. 

### Steps to reproduce

1. Disable linting in your `biome.json` file with `  "linter": { "enabled": false }`.
2. In VSCode, Open a source file that contains code that violates one of your linting rules. Observe that the line is flagged.

### Expected behavior

Lines with style / linting issues are not flagged in the editor if the linter is not enabled.

### Does this issue occur when using the CLI directly?

Not sure / Not applicable

### Logs

```plain text

```
